tailnet/controllers.go

```diff
@@ -922,14 +975,13 @@ type tunnelUpdater struct {
 	client         WorkspaceUpdatesClient
 	coordCtrl      *TunnelSrcCoordController
 	dnsHostsSetter DNSHostsSetter
+	updateHandler  UpdatesHandler
 	ownerUsername  string
 	recvLoopDone   chan struct{}
 
-	// don't need the mutex since only manipulated by the recvLoop
-	workspaces map[uuid.UUID]*workspace
-
 	sync.Mutex
-	closed bool
+	workspaces map[uuid.UUID]*Workspace
```

Comment:
If this is protected by the mutex, I suggest renaming all the functions that assume the mutex is locked with a `Locked` suffix, like `allAgentIDsLocked`